### PR TITLE
Add a blog post introducing scalajs-bundler

### PR DESCRIPTION
**Pull request by julienrf**

### CODE REVIEW

File: blog/_posts/2016-10-19-scalajs-bundler.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+[Scala.js](https://www.scala-js.org/) projects that use [npm](https://www.npmjs.com/) packages.
+
+## Motivation
+
+npm is the most popular JavaScript package registry. How can Scala.js applications benefit from
+the many libraries published on npm? First, this requires to resolve these libraries, as well as
+their transitive dependencies, from the npm registry and to download them. Then, the artifacts
+need to be somehow “linked” with the Scala.js code. This latter step is more subtle than it seems
+because the linking process might vary according to the target execution environment. For instance,
+Node.js expects artifacts to be conform to the CommonJS format, whereas this format is not compatible
```
Comment (sjrd) [inline blog/_posts/2016-10-19-scalajs-bundler.md]: to ~~be~~ conform

File: blog/_posts/2016-10-19-scalajs-bundler.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+There are mainly two existing approaches to tackle these challenges: using
+[WebJars](http://www.webjars.org/) or combining two build systems (one for the Scala
+world and one for the npm world). Both of them require extra efforts from developers
+or have limitations (you can find more details about that in scalajs-bundler’s
+[documentation](https://scalacenter.github.io/scalajs-bundler/motivation.html)).
+
+## Vision, current state and future work
+
+The last release of Scala.js brought support for
```
Comment (sjrd) [inline blog/_posts/2016-10-19-scalajs-bundler.md]: latest

File: blog/_posts/2016-10-19-scalajs-bundler.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+layout: blog
+post-type: blog
+by: Julien Richard-Foy
+title: "scalajs-bundler has been released"
+---
+
+This week we have released a first version of [`scalajs-bundler`](https://scalacenter.github.io/scalajs-bundler), a tool aiming to provide an integrated developer experience for
+[Scala.js](https://www.scala-js.org/) projects that use [npm](https://www.npmjs.com/) packages.
+
+## Motivation
```
Comment (olafurpg) [inline blog/_posts/2016-10-19-scalajs-bundler.md]: The challenge? This section doesn't go so much into motivation.

File: blog/_posts/2016-10-19-scalajs-bundler.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+---
+layout: blog
+post-type: blog
+by: Julien Richard-Foy
+title: "scalajs-bundler has been released"
```
Comment (olafurpg) [inline blog/_posts/2016-10-19-scalajs-bundler.md]: "Use npm packages in Scala.js with scalajs-bundler"? The title is quite important for search engines so pick those words carefully!

**Comment by SethTisue** [comment]
Somewhere in the post, I think it would be wise to put this in the context of the advisory-board-accepted proposal it falls under (with a link). You don't need to give a lot of space to it, but I think it's important to continually remind people that the center is proposal-driven.

File: blog/_posts/2016-10-19-scalajs-bundler.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+---
+
+This week we have released a first version of [`scalajs-bundler`](https://scalacenter.github.io/scalajs-bundler), a tool aiming to provide an integrated developer experience for
```
Comment (jvican) [inline blog/_posts/2016-10-19-scalajs-bundler.md]: What about "This week we release a first version of the [`scalajs-bundler`](https://scalacenter.github.io/scalajs-bundler), a tool that improves the use of [npm]() packages for [Scala.js]() developers." ?

File: blog/_posts/2016-10-19-scalajs-bundler.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+[Scala.js](https://www.scala-js.org/) projects that use [npm](https://www.npmjs.com/) packages.
+
+## Motivation
+
+npm is the most popular JavaScript package registry. How can Scala.js applications benefit from
+the many libraries published on npm? First, this requires to resolve these libraries, as well as
+their transitive dependencies, from the npm registry and to download them. Then, the artifacts
+need to be somehow “linked” with the Scala.js code. This latter step is more subtle than it seems
```
Comment (jvican) [inline blog/_posts/2016-10-19-scalajs-bundler.md]: ... need to be ~~somehow~~ *linked* with the Scala.js code. This last step is subtler than it seems -- the linking process varies according to the target execution environment.

File: blog/_posts/2016-10-19-scalajs-bundler.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+or have limitations (you can find more details about that in scalajs-bundler’s
+[documentation](https://scalacenter.github.io/scalajs-bundler/motivation.html)).
+
+## Vision, current state and future work
+
+The last release of Scala.js brought support for
+[CommonJS modules](https://www.scala-js.org/doc/project/module.html), thus opening new
+possibilities.
+
+scalajs-bundler aims to make npm dependencies management as easy as JVM dependencies
```
Comment (jvican) [inline blog/_posts/2016-10-19-scalajs-bundler.md]: "**The** scalajs-bundler" sounds better :+1:

File: blog/_posts/2016-10-19-scalajs-bundler.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+management.
+
+Basically, the idea is that if an application needs to depend on an
```
Comment (jvican) [inline blog/_posts/2016-10-19-scalajs-bundler.md]: I would merge this paragraph with the previous one and reword it as:
"In a nutshell, if you need to depend on a npm package (`foo`), add the following line to your sbt build:"

File: blog/_posts/2016-10-19-scalajs-bundler.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+~~~
+
+And then the `run` and `test` sbt commands just work.
+
+The 0.1 release contains an sbt plugin that:
+
+- lets developers define their npm dependencies (as in the above example),
+- keeps track of transitive npm dependencies between Scala.js artifacts,
+- fetches these dependencies from the npm registry,
+- provides tasks to bundle the application into a single artifact executable by Web browsers.
```
Comment (jvican) [inline blog/_posts/2016-10-19-scalajs-bundler.md]: s/Web/web ?

**Comment by julienrf** [comment]
Thank you so much for your feedback. I just pushed some commits applying your suggestions.

**Comment by heathermiller** [comment]
LGTM, thanks @julienrf!!

**Pull request merged**